### PR TITLE
PR for feature branch

### PR DESCRIPTION
**Pull request by praz99**
In this project we

- added user sessions.

- generated controller for user, event, and event attendance.

- created associations between users and events.

- added RSpec test for model validations and associations.

- added RSpec and Capybara for testing features and integrations.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  create_table "users", force: :cascade do |t|
+    t.string "name"
+    t.string "username"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
+
+  add_foreign_key "event_attendings", "events", column: "attended_events_id"
+  add_foreign_key "event_attendings", "users", column: "event_attendee_id"
```
Comment (lekegitrepo) [inline db/schema.rb]: - The following columns of the database should be index
  - creator_id
  - event_attendings

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
+      <li class="text-danger"><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
```
Comment (lekegitrepo) [inline app/views/shared/_error_messages.html.erb]: - View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
- There should be absolutely NO logic in views.

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        <nav class="navbar navbar-light bg-light justify-content-between">
+          <%= link_to current_user.name, user_path(current_user) %>
+          <%= link_to "log out", logout_path, class: 'mx-2', method: :delete %>
+        </nav>
+      <% else %>
+        <nav class="navbar navbar-light bg-light justify-content-between">
+          <%= link_to "Log in", login_path, class: 'mx-2' %>
+          <%= link_to "Sign up", signup_path, class: 'mx-2' %>
+        </nav>
+      <% end %>
```
Comment (lekegitrepo) [inline app/views/layouts/_header.html.erb]: - View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
- There should be absolutely NO logic in views.

File: app/views/event_attendings/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<h1>Attend Event</h1>
+<%= form_for [@event, @event.event_attendings.build] do |f| %>
+<!--<%= f.collection_check_boxes :event_attendee_id, User.all, :id, :username %>-->
```
Comment (lekegitrepo) [inline app/views/event_attendings/new.html.erb]: - It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle:
#### Hi @praz99 and @okikiola11 :clap: :clap:
#### You have done a good job so far, however, there are still some requirements you need to fix before this project can be approved. Check my comments below.

**_- The event's show should page displays the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/login_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**_- The user should provide email ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_can_create_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**_Check the remaining comments below :point_down:_** 

#### _Reviewed by Temitayo Adeleke._

File: app/controllers/event_attendings_controller.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+#   end
+
+#   def show
+#     @event_attending = EventAttending.find(params[:id])
+#   end
+
+#   def att_params
+#     params.raquire(:event_attending).permit(:event_attendee_id, :attended_events_id)
+#   end
+# end
```
Comment (lekegitrepo) [inline app/controllers/event_attendings_controller.rb]: - It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

**Review by rOluochKe — APPROVED** [review]
Hello Team
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project, good luck with the next one.
- Happy coding!

**Pull request merged**